src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -497,6 +499,22 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
             return count;
         }
 
+        private static class UnderlyingArrayHelper<T>
+        {
+            public static readonly Func<object, object> GetValue =
+                typeof(ImmutableArray<T>).GetField("array",
+                    BindingFlags.Instance | BindingFlags.NonPublic).GetValue;
```

Comment:
>BindingFlags.Instance | BindingFlags.NonPublic).GetValue; [](start = 20, length = 57)

Is it safe to assume that the field will always be found?

Reply:
It looks like so far we only use this API for ```ImmutableArray<int>``` and ```ImmutableArray<byte>```. Why not use specialized implementation for them as in ImmutableArrayInterop.cs and then we could add more specialization when more types need handling?

---
In reply to: [241927944](https://github.com/dotnet/roslyn/pull/31785#discussion_r241927944) [](ancestors = 241927944)